dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -327,32 +378,42 @@ public void ContentFallbackFalse_PageFallbackTrue_PageEnglish_ViewEnglishContent
                         .setPageMode(PageMode.LIVE)
                         .build(),
                 mockRequest, mockResponse);
-        Assert.assertTrue("ESP = "+html , html.contains("content3content2Spa"));
+        Assert.assertTrue("ESP = "+html , html.contains("content3Spacontent2Spa"));
+    }
+
+    @NotNull
+    private HTMLPageAsset createHtmlPageAsset(
+            final Template template,
+            final String pageName,
+            final long languageId)
+
+            throws DotSecurityException, DotDataException {
+        final HTMLPageAsset pageEnglishVersion = new HTMLPageDataGen(folder,template).languageId(languageId).pageURL(pageName).title(pageName).nextPersisted();
+        pageEnglishVersion.setIndexPolicy(IndexPolicy.WAIT_FOR);
+        pageEnglishVersion.setIndexPolicyDependencies(IndexPolicy.WAIT_FOR);
+        pageEnglishVersion.setBoolProperty(Contentlet.IS_TEST_MODE, true);
+        contentletAPI.publish(pageEnglishVersion, systemUser, false);
+        addAnonymousPermissions(pageEnglishVersion);
+        return pageEnglishVersion;
     }
 
     /**
-     * ContentFallback False
-     * PageFallback True
-     *
-     * Page English & Spanish
-     *
-     * English -> 1 & 2
-     * Spanish -> 2 & 3
-     *
+     * Method to test: {@link HTMLPageAssetRenderedAPI#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
+     * When: DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE is set to false and DEFAULT_PAGE_TO_DEFAULT_LANGUAGE is set to true
+     *       And the page have version in ENG and ESP
+     *       And the page have tree content, where: content1 is just in ENG version, content2 is in ENG and ESP version, content 3 is just in ESP version
+     * Should: If the page is requests in ENG version it should be render with content1 and content2
+     *         If the page is requests in ESP version it should be render with content3 and content2 (both in ESP version)
      */
     @Test
-    public void ContentFallbackFalse_PageFallbackTrue_PageEnglishAndSpanish_ViewEnglishContent1And2_ViewSpanishContent2And3() throws Exception{
+    @UseDataProvider("cases")
+    public void ContentFallbackFalse_PageFallbackTrue_PageEnglishAndSpanish_ViewEnglishContent1And2_ViewSpanishContent2And3(final Container container, final Template template) throws Exception{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5067852)